src/perform_step/adams_bashforth_moulton_perform_step.jl

```diff
@@ -544,3 +544,150 @@ end
   end
   f(k, u, p, t+dt)
 end
+
+# Variable Step Size Multistep Methods
+
+function initialize!(integrator,cache::VSA3ConstantCache)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+@muladd function perform_step!(integrator,cache::VSA3ConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack k2,k3,grid_points,ϕstar_nm1,k,order,idx,success,tab = cache
+  
+  k1 = integrator.fsalfirst
+  cnt = integrator.iter
+
+  if cnt <= 3
+    idx = cnt
+    grid_points[idx] = t
```

Comment:
I think that using a grid of `t`s is not a good idea here, since you are going to compute `grid_points[last_idx] - grid_points[last_idx-i]` and [catastrophic cancellation](https://en.wikipedia.org/wiki/Loss_of_significance) will occur under floating point arithmetic. Store `dt`s instead. By the way, I used the variable `tau` to store past `dt`s. https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/caches/adams_nordsieck_caches.jl#L11